### PR TITLE
Replace macOS native osuTK reflection

### PR DESCRIPTION
**Pull request by swoolcock**
Replaces the reflected osuTK calls in `MacOS.Native` with real implementations.  Required as part of dropping osuTK.  Implementations loosely borrowed from osuTK.

Note that the `FoundationLibrary` reference is not required and has been removed.

Related to #2448

### CODE REVIEW

File: osu.Framework/Platform/MacOS/Native/Cocoa.cs [hunk 1]
```diff
@@ -61,30 +69,45 @@ internal static class Cocoa
...
         public static extern bool CGCursorIsVisible();
 
         [DllImport(LIB_CORE_GRAPHICS, EntryPoint = "CGEventSourceFlagsState")]
         public static extern ulong CGEventSourceFlagsState(int stateID);
 
+        [DllImport(LIB_DL)]
+        private static extern IntPtr dlsym(IntPtr handle, string name);
+
+        [DllImport(LIB_DL)]
+        private static extern IntPtr dlopen(string fileName, int flags);
```
Comment (huoyaoyuan) [inline osu.Framework/Platform/MacOS/Native/Cocoa.cs]: Don't P/Invoke on this - use `NativeLibrary.Load` instead.

File: osu.Framework/Platform/MacOS/Native/Cocoa.cs [hunk 2]
```diff
@@ -61,30 +69,45 @@ internal static class Cocoa
...
-        public static IntPtr FoundationLibrary;
 
         [DllImport(LIB_CORE_GRAPHICS, EntryPoint = "CGCursorIsVisible")]
         public static extern bool CGCursorIsVisible();
 
         [DllImport(LIB_CORE_GRAPHICS, EntryPoint = "CGEventSourceFlagsState")]
         public static extern ulong CGEventSourceFlagsState(int stateID);
 
+        [DllImport(LIB_DL)]
+        private static extern IntPtr dlsym(IntPtr handle, string name);
```
Comment (huoyaoyuan) [inline osu.Framework/Platform/MacOS/Native/Cocoa.cs]: Use `NativeLibrary.GetExport/TryGetExport`.

**Comment by swoolcock** [comment]
Reason?

**Comment by huoyaoyuan** [comment]
The reason is direct: if there's framework provided functionality, always use it instead of directly calling into the OS.
Although this is not portable code, use `NativeLibrary` on every platform still looks cleaner.

**Comment by huoyaoyuan** [comment]
Maybe I've misunderstood something. `TryGetExport` is mainly for loading function exports, not sure about string constants. But it at least worth a try.

**Comment by swoolcock** [comment]
I’m planning on replacing most of this native code with `NativeLibraryLoader` anyway.

https://github.com/mellinoe/nativelibraryloader

**Comment by huoyaoyuan** [comment]
I didn't qualify it, but I meant `System.Runtime.InteropServices.NativeLibrary`. Please prefer it when it's suitable. .NET Core has worked a lot for cross-platform interop.

**Comment by smoogipoo** [comment]
I'm not interested in increasing the complexity of implementation by using a new tool which may or may not have its own problems. This is just intended to take us further away from osuTK.

**Pull request merged**